iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -443,33 +567,31 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             if (Logging.IsEnabled)
                 Logging.Enter(this, cancellationToken, nameof(CloseAsync));
 
-            if (!_isOpen)
-            {
-                // Already closed so gracefully exit, instead of throw.
-                return;
-            }
-
-            await _handlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
-
             try
             {
-                if (!_isOpen)
+                if (GetClientTransportStatus() == ClientTransportStatus.Closed)
                 {
                     // Already closed so gracefully exit, instead of throw.
                     return;
                 }
+                SetClientTransportStatus(ClientTransportStatus.Closing);
 
-                _handleDisconnectCts.Cancel();
-                await base.CloseAsync(cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    _cancelPendingOperationsCts?.Cancel();
+                    _handleDisconnectCts?.Cancel();
+                    await base.CloseAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
@drwill-ms We had discussed encapsulating the `CloseAync()` within a try-catch block so that close failures are suppressed. However, as I was testing this flow I had another thought. Since we do allow open-close-open flow, I wanted to avoid suppressing any exceptions in here as failure to cleanly close the transport layer might result in issues down the line if subsequently `OpenAsync()` is called. For this reason, I wanted to leave this as-is and evaluate this in the future if we get feedback from customers on this flow.